### DISCUSSION TITLE
How to chain where, limit, and order_by without a select statement?

### DISCUSSION THREAD

**Discussion by seperman** [sqlalchemy/sqlalchemy#11472] [Usage Questions] [answered]
Hello,
I have a chain of where clauses, limit, order_by, offset, etc.
After creating the chain, I would like to decide on the select statement.
Whether it is `select(MyModel)` or `select(func.count(MyModel.id))`, or something else.
Use case: Use the chain of filters to select the count of results and also return a subset of them after adding more where clauses to the chain.

How can I do that? Thanks.

**Comment by CaselIT** [comment, marked as answer]
Hi,

You will need to use a select, you can replace the select clause using https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Select.with_only_columns

**Comment by seperman** [comment]
Hi @CaselIT 
Thanks for the prompt response. Ok, I will check it out. I ended up separating the chain components and re-assembling them for each select statement.
